sound/soc/sof/intel/intel-client.c

```diff
@@ -10,6 +10,21 @@
 #include "../sof-client.h"
 #include "intel-client.h"
 
+void intel_unregister_client(struct snd_sof_dev *sdev, const char *name)
+{
+	struct sof_client_dev *cdev, *_cdev;
+
+	mutex_lock(&sdev->client_mutex);
+
+	list_for_each_entry_safe(cdev, _cdev, &sdev->client_list, list) {
+		if (!strcmp(cdev->auxdev.name, name))
+			sof_client_dev_unregister(cdev);
+	}
+
+	mutex_unlock(&sdev->client_mutex);
+}
+EXPORT_SYMBOL_NS_GPL(intel_unregister_client, SND_SOC_SOF_INTEL_CLIENT);
+
```

Comment:
this should be squashed in the earlier patch where we initially used 'ipc_test'.